### PR TITLE
Containers

### PR DESCRIPTION
**Pull request by Parth**

### CODE REVIEW

File: server/Cargo.toml [hunk 1]
```diff
@@ -6,15 +6,15 @@ edition = "2018"
...
+env_logger = "0.7.1"
+hyper = "0.13.5"
+lockbook_core = { path = "../core" }
+log = "0.4.8"
 openssl = { version = "0.10", features = ["vendored"] }
-tokio-postgres = "0.5.4"
-rust-s3 = "0.18.6"
+postgres-openssl = "0.3.0"
+rsa = { version = "^0.2", features = ["serde1"] }
+rust-s3 = { version = "0.22.3", features = ["path-style"] }
```
Comment (tvanderstad) [inline server/Cargo.toml:16]: note: new rust-s3 version required for connecting to custom URLs for testing, version includes breaking changes such as making everything async

File: test/tests/change_file_content.rs [hunk 2]
```diff
@@ -8,7 +6,7 @@ use lockbook_core::model::api::NewAccountRequest;
 use lockbook_core::model::api::{ChangeFileContentError, ChangeFileContentRequest};
 
 #[macro_use]
-mod utils;
+pub mod utils;
```
Comment (tvanderstad) [inline test/tests/change_file_content.rs:9]: note: tests are built as their own modules; shared functions that are unused in any of the tests produced unused warnings even if used in other tests. Making this public suppresses those warnings

File: server/src/config.rs [hunk 3]
```diff
@@ -27,21 +32,28 @@ pub struct Config {
 pub fn config() -> Config {
     Config {
         index_db_config: IndexDbConfig {
-            user: env!("INDEX_DB_CONFIG_USER"),
-            pass: env!("INDEX_DB_CONFIG_PASS"),
-            host: env!("INDEX_DB_CONFIG_HOST"),
-            port: env!("INDEX_DB_CONFIG_PORT").parse().unwrap(),
-            db: env!("INDEX_DB_CONFIG_DB"),
-            cert: env!("INDEX_DB_CONFIG_CERT"),
+            host: env_or_panic("INDEX_DB_HOST"),
```
Comment (tvanderstad) [inline server/src/config.rs:35]: note: config now done at runtime

File: containers/dummy.rs [hunk 4]
```diff
@@ -0,0 +1,5 @@
+pub fn main() {}
+
+/// Intentionally leave out pub to generate a warning so you know when this file is being used
+/// Read more here: https://blog.mgattozzi.dev/caching-rust-docker-builds/
+fn you_are_using_dummy_rs_for_docker_cache_grep_me() {}
```
Comment (tvanderstad) [inline containers/dummy.rs:5]: @Parth do we need the grep_me? unscalable pattern

File: .github/workflows/build.yml [hunk 5]
```diff
@@ -48,46 +41,91 @@ jobs:
   Server:
     runs-on: ubuntu-latest
     env:
-      working-directory: ./server
-      INDEX_DB_CONFIG_USER: ${{ secrets.INDEX_DB_CONFIG_USER }}
```
Comment (tvanderstad) [inline .github/workflows/build.yml:52]: note: no longer need secrets for tests, config now stored in containers/test.env (message me for qa secrets)

File: .github/workflows/build.yml [hunk 6]
```diff
@@ -48,46 +41,91 @@ jobs:
...
+      run: make cli_push
+    - name: Notify Slack
+      uses: 8398a7/action-slack@v3
+      with:
+        status: ${{ job.status }}
+        text: Build Cli - ${{ job.status }}
+      env:
+        SLACK_WEBHOOK_URL: ${{ secrets.SLACK_WEBHOOK }}
+      if: always()
+  Test:
```
Comment (tvanderstad) [inline .github/workflows/build.yml:99]: @Parth please review this part carefully

File: containers/dummy.rs (hunk 4 above)
Comment (Parth) [reply containers/dummy.rs:5]: your call

**Pull request merged**